src/sidebar/components/TopBar.js

```diff
@@ -21,19 +21,21 @@ import UserMenu from './UserMenu';
 
 /**
  * @typedef TopBarProps
- * @prop {AuthState} [auth]
- * @prop {Bridge} bridge
- * @prop {boolean} [isSidebar] - Flag indicating whether the app is the sidebar or a top-level page.
- * @prop {() => any} [onLogin] - Callback invoked when user clicks "Login" button.
- * @prop {() => any} [onLogout] - Callback invoked when user clicks "Logout" action in account menu.
- * @prop {() => any} [onSignUp] - Callback invoked when user clicks "Sign up" button.
- * @prop {MergedConfig} [settings]
- * @prop {Object} [streamer]
+ * @prop {AuthState} auth
+ * @prop {Bridge} [bridge]
+ * @prop {boolean} isSidebar - Flag indicating whether the app is the sidebar or a top-level page.
+ * @prop {() => any} onLogin - Callback invoked when user clicks "Login" button.
+ * @prop {() => any} onLogout - Callback invoked when user clicks "Logout" action in account menu.
+ * @prop {() => any} onSignUp - Callback invoked when user clicks "Sign up" button.
+ * @prop {MergedConfig} settings
+ * @prop {import('../services/streamer').StreamerService} streamer
  */
```

Comment:
Update the type of `streamer` and correct the optionality of various other properties. It was incorrect but not noticed because the `TopBarProps` type was not actually used.